src/FakeItEasy/Creation/DummyValueResolver.cs

```diff
@@ -66,12 +65,12 @@ private bool TryResolveDummyValueWithAllAvailableStrategies(DummyCreationSession
             {
                 if (strategy.TryCreateDummyValue(session, typeOfDummy, out result))
                 {
-                    this.strategyCache.Add(typeOfDummy, strategy);
+                    this.strategyCache.TryAdd(typeOfDummy, strategy);

```

Comment:
You mean if it throws? It's pretty unlikely. `typeOfDummy` isn't `null`, and we're unlikely to have `Int32.MaxValue` items…

If it returns `false`, it just means there was already a strategy registered. We don't mind competing threads finding the strategy and inserting one first. Unless you'd rather we were super strict and locked out other threads. I can try that, although the method we have to pass to `GetOrAdd` is kind of complicated, I think.